docs/applications.md

```diff
@@ -80,7 +80,7 @@ exceptions that occur within the application:
 ### Storing state on the app instance
 
 You can store arbitrary extra state on the application instance, using the
-generic `app.state` attribute.
+generic `app.state` attribute.  This can then be accessed in middleware or endpoints via `request.app`.
```

Comment:
Rather than having `request.app` mentioned specifically wrt. `request.app.state`, I'd rather we had a one or two sentence section "Accessing the app instance", which just notes that you can either import the application instance, or that it is also available as `request.app`. (Makes a bit more sense to me that way around). Make sense?

Reply:
Yeah makes sense.

Reply:
It could make sense if you want to access the `app.state` from a place where you don't have access to a request (e.g., in a repeating task that runs periodically).

(It wouldn't necessarily require importing, e.g. if the task was configured in the same file as the app was created, but it would still involve accessing the app instance without needing to read it off of a request.)

Reply:
Updated version pushed, but mentions the use of `request.app` first.  Feel free to edit directly if you prefer different wording @tomchristie.